pkg/printers/sarif.go

```diff
@@ -0,0 +1,212 @@
+// Additionally, there are custom GJSON modifiers, which will post-process expression results. Currently,
+// the gjson.ListModifier, gjson.TextModifier and gjson.BoolReplaceModifier are available, see their documentation on
+// usage and GJSON's syntax expression to read more about modifiers.
+//
+// The following example illustrates a JSON compatible structure and an example for the map of JSON path expressions
+// to generate a sarif report.
+//
+// JSON structure:
+//
+//		type data struct {
+//					Violations []violation `json:"violations`
+//		}
+//
+//		type violation struct {
+//		      Id          string `json:"id"`
+//	          Description string `json:"description"`
+//	          Reason      string `json:"reason"`
+//	          Severity    string `json:"severity"`
+//		}
+//
+// Example:
+//
+//	 expressions := map[string] {
+//			RuleJSONPathExpressionKey: "violations.#.id",
+//			DescriptionJSONPathExpressionKey: "violations.#.description",
+//		    HelpJSONPathExpressionKey: "violations.#.reason",
+//		    SeverityJSONPathExpressionKey: "violations.#.severity",
+//		}
+//
+// For an example sarif report, see testdata/sarif_report.json.
+//
+// Advanced usages:
+// For constructing multiline help messages with values from different JSON fields, see roxctl/image/check/check.go
+// as an example usage of constructing 1) a rule ID with multiple values using gjson.TextModifier and 2) creating the
+// multiline help text via gjson.TextModifier.
+func NewSarifPrinter(jsonPathExpressions map[string]string, entity string, reportType string) *SarifPrinter {
+	return &SarifPrinter{jsonPathExpressions: jsonPathExpressions, entity: entity, reportType: reportType}
+}
+
+// Print will create a sarif report from the given object and write the output to the given io.Writer.
+func (s *SarifPrinter) Print(object interface{}, out io.Writer) error {
+	sarifEntries, err := sarifEntriesFromJSONObject(object, s.jsonPathExpressions)
+	if err != nil {
+		return err
+	}
+
+	run := sarif.NewRunWithInformationURI("roxctl", "https://github.com/stackrox/stackrox")
+	// Set the version information and the full name on the tool driver.
+	run.Tool.Driver.
+		WithVersion(version.GetMainVersion()).
+		WithFullName("roxctl command line utility")
+
+	report, err := sarif.New(sarif.Version210)
+	if err != nil {
+		return errors.Wrap(err, "creating sarif report")
+	}
+
+	for _, entry := range sarifEntries {
+		addEntry(run, entry, s.entity, s.reportType)
+	}
+
+	report.AddRun(run)
+	return report.PrettyWrite(out)
+}
+
+func addEntry(run *sarif.Run, entry sarifEntry, entity string, name string) {
+	run.AddRule(entry.ruleID).
+		WithName(name).
+		WithShortDescription(sarif.NewMultiformatMessageString(entry.description)).
+		WithFullDescription(sarif.NewMultiformatMessageString(entry.description)).
+		WithHelp(sarif.NewMultiformatMessageString(entry.help))
+
+	run.AddResult(sarif.NewRuleResult(entry.ruleID).
```

Comment:
Hm, yeah I guess we could, however GitHub already has the `Tool` information that's filled out, so not sure whether this would be too much, but I'm happy with the change.